test_ansible_shell.yml

```diff
@@ -0,0 +1,16 @@
+---
+- name: Verify that ansible can correctly pass arguments to ansible in shell
+  hosts: all
+  gather_facts: False
+  tasks:
+    - name: Write inventory file
+      blockinfile:
+        path: '{{ playbook_dir }}/inventory.ansible.shell'
```

Comment:
I think this makes sense, but what is the practical effect when ran inside of Tower? Does this actually get saved some some other directory and get cleaned up at the end of the job?

The playbook directory (IIRC) is where the project checkout is, so I'm curious what happens when you leave files lying around there, and if it's hazardous in the event that bubblewrap is turned off. One thing I'm curious about is if you subsequently update the project, does it detect uncommitted changes? What does it do in that event?

Reply:
I've had to use this pattern before in some unusual circumstances (customer wanted to store a CMDB in a git repo and clone it for use it in playbooks while capturing some metadata about the current revision, but also didn't want to disable bubblewrap)

This will write the file into the project checkout directory. When you do this, you have to make sure that you've got `scm_clean: True` or `scm_delete_on_update` , otherwise the next checkout will fail. Since we never recycle projects for our tests, this won't cause any problems for us, and we can leave bubblewrap enabled.